### ISSUE TITLE
Suggestion: Allow components to render in the "el"

### ISSUE THREAD

**Issue by cfenzo** [ractivejs/ractive#1131]
Right now, if you set the `el` option for a component, it will render in the element specified (like `body`, or `#main`), but crash the whole Ractive instance:

``` javascript
Ractive.components.modal = Ractive.extend({
  el:'body',
  append:true
});
// gives: Uncaught Error: You cannot call ractive.render() on an already rendered instance! Call ractive.unrender() first 
```

I'm not sure if this is a bug, or intended behavior..

_Why would you want to render anywhere but where the tag is?_ one might say..
In some cases, like with a modal or a dropdown, it would be very nice to be able to render the component outside the template (like appended to the body), or outside a list (could be done in more ways).

Rendered Ractive components can be moved outside the template/into any element, by using a combination of `complete`, `detach` and `insert`:

``` javascript
Ractive.components.Modal = Ractive.extend({
    template:'<div class="modal">{{>content}}</div>',
    data:{
      el:false
    },
    append:true,
    complete:function(){
      this.reattach();
    },
    init:function(){
      this.observe('el',function(n,o){
        this.reattach();
      },{init:false});
    },
    reattach:function(){
      if(this.get('el')){
        this.detach();
        this.insert(this.get('el'));
      }
    }
  });
```

(jsbin example)[http://jsbin.com/xusez/19]
But this workaround will first render the component inside the template (and run all intro animations), and then move the rendered component outside to the specified element. Which can cause glitches..

I hope "rendering components in the element referenced by the `el` option" is a feature worth adding (it looks like it's doable), maybe there should be an option to toggle the behavior?

But if components shouldn't be able to render in the `el`, it would be nice if the `el` option where ignored for components.

**Comment by evs-chris** [comment]
In 0.9 builds, you can now attach external instances, that can handle rendering themselves or let the parent instance handle it. That _should_ cover cases like this, where you need manual control but still want to maintain the parent -> child relationship.
